### PR TITLE
[openvino] update to 2024.4.0

### PR DESCRIPTION
**Pull request by culhatsker**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] When updating the upstream version, the `"port-version"` is reset (removed from `vcpkg.json`).
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by culhatsker** [comment]
I disabled arm&android configuration cause latest ComputeLibrary has issues with it.
See https://github.com/ARM-software/ComputeLibrary/issues/1141 for details

File: ports/openvino/vcpkg.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   "homepage": "https://github.com/openvinotoolkit/openvino",
   "documentation": "https://docs.openvino.ai/latest/index.html",
   "license": "Apache-2.0",
-  "supports": "!uwp & !x86",
+  "supports": "!uwp & !x86 & !(android & arm)",
```
Comment (dg0yt) [inline ports/openvino/vcpkg.json:17]:
```suggestion
  "supports": "!uwp & !x86 & !(android & arm32)",
```

IIUC arm64 isn't affected.

Comment (culhatsker) [reply ports/openvino/vcpkg.json]: Oh, you're right. I assumed the keyword would be arm for non arm64

**Comment by JonLiu1993** [comment]
@culhatsker, Thanks for your PR, when I tested feature by command ./vcpkg install `openvino[*]:x64-windows`. I get this error, please take a look:
````
CMake Error at thirdparty/gflags/CMakeLists.txt:16 (add_subdirectory):
  The source directory

    F:/Feature-test/openvino/buildtrees/openvino/src/2024.4.0-73dc795f05.clean/thirdparty/gflags/gflags

  does not contain a CMakeLists.txt file.
````

**Comment by ilya-lavrenov** [comment]
> @culhatsker, Thanks for your PR, when I tested feature by command ./vcpkg install `openvino[*]:x64-windows`. I get this error, please take a look:
> 
> ```
> CMake Error at thirdparty/gflags/CMakeLists.txt:16 (add_subdirectory):
>   The source directory
> 
>     F:/Feature-test/openvino/buildtrees/openvino/src/2024.4.0-73dc795f05.clean/thirdparty/gflags/gflags
> 
>   does not contain a CMakeLists.txt file.
> ```

need to add cmake option `ENABLE_INTEL_NPU_INTERNAL=OFF`

**Comment by culhatsker** [comment]
Ok, I see that this issue also happens on `x64-linux-dynamic` triplet and I need to add the `ENABLE_INTEL_NPU_INTERNAL=OFF` option when npu feature is enabled.
I never checked all features cause I assumed that CI does this check for port updates *shrugs*

**Comment by dg0yt** [comment]
> thirdparty/gflags/CMakeLists.txt:16 (add_subdirectory):

If somethings needs gflags, it should take it from vcpkg.

**Comment by ilya-lavrenov** [comment]
> > thirdparty/gflags/CMakeLists.txt:16 (add_subdirectory):
> 
> If somethings needs gflags, it should take it from vcpkg.

vcpkg enabled components do not require gflags.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
All features are tested successfully in `x64-windows` triplet.

**Comment by culhatsker** [comment]
@JonLiu1993 may I ask you to check it again? I wasn't able to check the fix on windows platform, but for a similar triplet in linux the problem doesn't reproduce anymore.

**Pull request merged**